### ISSUE TITLE
Error on executing IpAdapterUnifiedLoader

### ISSUE THREAD

**Issue by uttams0lanki** [cubiq/ComfyUI_IPAdapter_plus#386]
I've got the clip visions renamed and the models in the ipadapter model yet I've been struggling with this issue,
what am I missing

![Screenshot (1) copy](https://github.com/cubiq/ComfyUI_IPAdapter_plus/assets/24311166/fbe86db9-5108-4ccc-b2b2-4be998ad26d6)

**Comment by cubiq** [comment]
you have very few ipadapter models, you are probably trying to load one that you don't have

**Comment by uttams0lanki** [comment]
![Screenshot (1) copy](https://github.com/cubiq/ComfyUI_IPAdapter_plus/assets/24311166/2ba321b3-ebd2-4acb-b9e1-c8de258cbac1)
@cubiq So I was using the ipadapter_potrait.json example file, i've downloaded all the required models, kept it in both the ipdapter folder and the faceid folder :\ I'm still getting the same error, does "clip missing: ['clip_l.logit_scale', 'clip_l.transformer.text_projection.weight']" has to do anything with it, if yes how does one fixes it

**Comment by Conmiro** [comment]
I was having the same issue using StabilityMatrix package manager to manage ComfyUI. For some reason, it doesn't detect the ipadatpter folder you create inside of ComfyUI/models. I had to put the IpAdapter files in `\AppData\Roaming\StabilityMatrix\Models` instead.

Looking here: https://github.com/comfyanonymous/ComfyUI/blob/master/folder_paths.py , I'm actually not sure how the added ipadapter folder would ever get picked up by the fodler_paths module, even when not using StabilityMatrix. I added a print statement inside ipadapter_plus utils.py:

```
def get_ipadapter_file(preset, is_sdxl):
    preset = preset.lower()
    ipadapter_list = folder_paths.get_filename_list("ipadapter")
    print(ipadapter_list)
```
The was an empty array no matter what files existed in my ipadapter folder.

**Comment by uttams0lanki** [comment]
![Capture](https://github.com/cubiq/ComfyUI_IPAdapter_plus/assets/24311166/11ddf659-b511-4861-9543-627ad2cf2ac1)

@Conmiro Thank you, but I'm not using StabilityMatrix, but my issue got fixed once I added the following line to my folder_paths.py file, weirdly every time I update my ComfyUI I have to repeat the process.

folder_names_and_paths["ipadapter"] = ([os.path.join(models_dir, "ipadapter")], ["folder"]) to

**Comment by cubiq** [comment]
that shouldn't be needed, the source of the problem is somewhere else

**Comment by Conmiro** [comment]
I agree. Since OP isn't using StabilityMatrix or anything similar, that mapping shouldn't exist at all and the code in IPAdapter should execute and add the entry for ipadapter folder to folder_names_and_paths. Glad OP has a workaround at least, but curious if they ever figure out root cause.

**Comment by cybearvision** [comment]
I have exactly the same problem as OP and not sure what is the work around. 

> ![Capture](https://private-user-images.githubusercontent.com/24311166/320949183-11ddf659-b511-4861-9543-627ad2cf2ac1.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM1OTIwOTksIm5iZiI6MTcxMzU5MTc5OSwicGF0aCI6Ii8yNDMxMTE2Ni8zMjA5NDkxODMtMTFkZGY2NTktYjUxMS00ODYxLTk1NDMtNjI3YWQyY2YyYWMxLlBORz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDIwVDA1NDMxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVmZGFlMTNjMzY3NGIyNGJkYWE3ZDliNjNhZTMyMTQ5NjRiYjQ0YWE3NWJhZTc4Yzc2NDU4NDUxN2Y5ZTJkZjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.N7gJ3jSoNQKsyVYnKpSOOVW9cXHcUCGRd6Hjh17K_xI)
> 
> @Conmiro Thank you, but I'm not using StabilityMatrix, but my issue got fixed once I added the following line to my folder_paths.py file, weirdly every time I update my ComfyUI I have to repeat the process.
> 
> folder_names_and_paths["ipadapter"] = ([os.path.join(models_dir, "ipadapter")], ["folder"]) to

Tried this and it didn't work for me.

**Comment by felix9339** [comment]
> ![Capture](https://private-user-images.githubusercontent.com/24311166/320949183-11ddf659-b511-4861-9543-627ad2cf2ac1.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc4MjMxMzAsIm5iZiI6MTcxNzgyMjgzMCwicGF0aCI6Ii8yNDMxMTE2Ni8zMjA5NDkxODMtMTFkZGY2NTktYjUxMS00ODYxLTk1NDMtNjI3YWQyY2YyYWMxLlBORz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA4VDA1MDAzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAxYTI1YmVjNTEzZmZjM2Q1YjZhOWM2NGY5M2MzMmM4ZTczZTAyN2JlNDQwM2ZlZjBkYzEyZWJmOGU2MTA2MjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.oPWlyMbHRwB5NERiWXMd1FpI1cLC4b1ISnQ-vW31PkA)
> 
> @Conmiro Thank you, but I'm not using StabilityMatrix, but my issue got fixed once I added the following line to my folder_paths.py file, weirdly every time I update my ComfyUI I have to repeat the process.
> 
> folder_names_and_paths["ipadapter"] = ([os.path.join(models_dir, "ipadapter")], ["folder"]) to

Hello can someone tell me how to do this process? I have zero knowledge of coding.I want to add the above line to my folder_paths.py file. I have no idea what to do..

**Comment by mariorodriguespt** [comment]
Also suffering from this issue

**Comment by samansalami** [comment]
having same problem, anyone has any solution ? PLEASEEEEE
